frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/nodes-selection-list.tsx

```diff
@@ -50,87 +50,51 @@ const setTableHeader = () => {
 
-const createNodeUIDMap = (nodes: NodeKind[]): NodeUIDMap =>
-  nodes.reduce((nodeUIDMap: NodeUIDMap, node: NodeKind) => {
-    const uid = getUID(node);
-    nodeUIDMap[uid] = node;
-    return nodeUIDMap;
-  }, {});
+const getRows: GetRows = ({ componentProps, customData }) => {
+  const { data: filteredData } = componentProps;
+  const { selectedNodes, setVisibleRows, visibleRows } = customData;
 
-const createRowUIDMap = (nodeUIDMap: NodeUIDMap, rows: RowUIDMap): RowUIDMap =>
-  Object.keys(nodeUIDMap).reduce((rowUIDMap, uid: string) => {
-    const node = nodeUIDMap[uid];
-    const nodeName = getName(node);
-    const nodeLocation = node.metadata.labels?.['failure-domain.beta.kubernetes.io/zone'] ?? '-';
-    const nodeCpuCapacity = getNodeCPUCapacity(node);
-    const nodeAllocatableMemory = getNodeAllocatableMemory(node);
+  const rows = filteredData.map((node: NodeKind) => {
+    const cpuSpec: string = getNodeCPUCapacity(node);
+    const memSpec: string = getNodeAllocatableMemory(node);
     const nodeTaints = node.spec?.taints?.length ?? 0;
-    const cells = [
+    const cells: IRow['cells'] = [
       {
-        title: <ResourceLink kind="Node" name={nodeName} title={uid} />,
+        title: <ResourceLink kind="Node" name={getName(node)} title={node.metadata.uid} />,
       },
       {
-        title: nodeCpuCapacity || '-',
+        title: `${humanizeCpuCores(cpuSpec).string || '-'}`,
       },
       {
-        title: humanizeBinaryBytes(convertToBaseValue(nodeAllocatableMemory)).string || '-',
+        title: humanizeBinaryBytes(convertToBaseValue(memSpec)).string ?? '-',
       },
       {
-        title: nodeLocation || '-',
+        title: node.metadata.labels?.['failure-domain.beta.kubernetes.io/zone'] ?? '-',
       },
       {
         title: pluralize(nodeTaints, 'taint'),
       },
     ];
-    rowUIDMap[uid] = {
+    return {
       cells,
-      selected: rows?.[uid]?.selected ?? false,
+      selected: getSelected(selectedNodes, node.metadata.uid),
       props: {
-        data: nodeUIDMap[uid],
-        uid,
+        id: node.metadata.uid,
       },
     };
-    return rowUIDMap;
-  }, {});
-
-const setTableRows: SetTableRows = ({ componentProps, customData }) => {
-  const { data: filteredData } = componentProps;
-  const { rows, setRows, allSelected, setAllSelected } = customData;
-
-  const nodeUIDMap = createNodeUIDMap(filteredData);
-  const rowUIDMap = createRowUIDMap(nodeUIDMap, rows);
-  const tableRows = Object.values(rowUIDMap);
+  });
 
-  if (allSelected !== null) {
-    /* Selecting and deselecting visible table rows */
-    Object.keys(rowUIDMap).forEach((uid) => (rowUIDMap[uid].selected = allSelected));
-    setRows({ ...rows, ...rowUIDMap });
-    setAllSelected(null);
-  } else if (!_.isEmpty(rowUIDMap) && stateShouldUpdate(rowUIDMap, rows)) {
-    setRows({ ...rows, ...rowUIDMap });
+  if (!_.isEqual(filteredData, visibleRows)) {
+    setVisibleRows(filteredData);
```

Comment:
Looks like it would be a better approach if we move the hook inside ListComponent and pass it `data`. With that approach, we could encapsulate this logic inside the hook as well.

Reply:
Then how are you planning to access selectedNodes/rows inside `getRows` method in Table component.

Reply:
Pass it via `customData`.

Reply:
We can't use hooks inside `ListComponent` as it is a class component. Hooks can only be called inside of the body of a function component.

Reply:
`ListComponent` is what you pass to a `ListPage`. And in this PR I see that you're using a Functional Component.

Reply:
I agree with @bipuladh . I think its worth a try to move the `useSelectList` and `visibleRows` management into the `ListComponent` (`NodeList` / `NodesSelectionList`). List component would keep the selected UIDs instead of the whole objects.
You could access the selected objects by adding something like `onSelectedRow` callback to ListComponent. The callbackwhich would pass back all selected objects on every selection change.

Reply:
Ack. I misunderstood the `ListComponent` with `ListPage` component which is a class base component.
Just a concern with this approach is that we need to maintain the selected nodes in both component i.e ListComponent and parent component, here its create-form.tsx. Hence, to avoid it, used custom hook in create-form.tsx itself.

Reply:
we already talked on chat, but just to recap here - parent component does not have to store the whole object just the parts that its interested in. In these cases, one of components is interested only in names, the other one in names and namespaces